test/test_linalg.py

```diff
@@ -7420,13 +7420,40 @@ def test_nuclear_norm_out(self, device, dtype):
 
     @skipCUDAIfNoMagma
     @skipCPUIfNoLapack
-    def test_geqrf(self, device):
-        a = torch.randn(5, 5, device=device)
-        b, c = torch.geqrf(a)
-        b_placeholder, c_placeholder = torch.empty_like(b), torch.empty_like(c)
-        torch.geqrf(a, out=(b_placeholder, c_placeholder))
-        self.assertEqual(b, b_placeholder)
-        self.assertEqual(c, c_placeholder)
+    @dtypes(torch.float32, torch.float64, torch.complex64, torch.complex128)
+    @dtypesIfCUDA(torch.float32, torch.float64)
+    def test_geqrf(self, device, dtype):
+
+        def run_test(shape):
+            # NumPy outputs the result of geqrf operation
+            # with mode = 'raw' of numpy.linalg.qr
+            # So we compare the implementation against NumPy
+            A = make_tensor(shape, dtype=dtype, device=device)
+
+            # NumPy's qr doesn't work with batched input
+            m, n = A.shape[-2:]
+            tau_size = "n" if m > n else "m"
+            np_dtype = A.cpu().numpy().dtype
+            ot = [np_dtype, np_dtype]
+            numpy_geqrf_batched = np.vectorize(
+                lambda x: np.linalg.qr(x, mode='raw'),
+                otypes=ot,
+                signature=f'(m,n)->(n,m),({tau_size})')
+
+            expected = numpy_geqrf_batched(A.cpu())
+            actual = torch.geqrf(A)
+
+            # NumPy's qr returns transposed result
+            self.assertEqual(expected[0].swapaxes(-2, -1), actual[0])
+            self.assertEqual(expected[1], actual[1])
+
+        batches = [(), (0, ), (2, ), (2, 1)]
+        ns = [5, 2, 0]
+        for batch, (m, n) in product(batches, product(ns, ns)):
+            # TODO: CUDA path doesn't work with batched or empty inputs
```

Comment:
Assert this fails instead of just continuing so when the behavior changes the test demands an update, too

Reply:
The test is updated in https://github.com/pytorch/pytorch/pull/56251. So can we leave this one as is? In other cases, I agree that it's better to assert failures and not skip the test.

Reply:
I'll add the assert.

Reply:
haha, really negotiating against yourself there